Tools-Override/CodeCoverage.targets

```diff
@@ -0,0 +1,132 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+    Code coverage package versions go here and in the test-runtime-packages.config
+  -->
+  <PropertyGroup>
+    <OpenCoverVersion>4.6.519</OpenCoverVersion>
+    <ReportGeneratorVersion>2.5.0</ReportGeneratorVersion>
+    <CoverallsUploaderVersion>1.4</CoverallsUploaderVersion>
+  </PropertyGroup>
+  
+  <!-- Coverage options -->
+  <PropertyGroup>
+    <CodeCoverageEnabled>false</CodeCoverageEnabled>
+    <CodeCoverageEnabled Condition="'$(SkipTests)' != 'true' and '$(OS)' == 'Windows_NT' and '$(Coverage)' == 'true' and '$(Performance)' != 'true'">true</CodeCoverageEnabled>
+    <CoverageReportDir Condition="'$(CoverageReportDir)' == ''">$(TestWorkingDir)coverage\</CoverageReportDir>
+
+    <!-- This targets file has two modes one for individual projects and one for all -->
+    <GenerateCodeCoverageReportForAll Condition="'$(GenerateCodeCoverageReportForAll)'==''">false</GenerateCodeCoverageReportForAll>
+    <CoverageEnabledForProject Condition="'$(GenerateCodeCoverageReportForAll)'=='true'">false</CoverageEnabledForProject>
+    <CoverageEnabledForProject Condition="'$(CoverageEnabledForProject)'=='' and '$(IsTestProject)'=='true'">$(CodeCoverageEnabled)</CoverageEnabledForProject>
+
+    <GenerateIndividualCoverageReport Condition="'$(BuildAllProjects)'!='true' and '$(CoverageEnabledForProject)'=='true'">true</GenerateIndividualCoverageReport>
+    <GenerateFullCoverageReport Condition="'$(GenerateCodeCoverageReportForAll)'=='true' and '$(CodeCoverageEnabled)'=='true' and '$(SkipFullCoverageReport)' != 'true'">true</GenerateFullCoverageReport>
+    <GenerateVisitedMethodsReport Condition="'$(GenerateCodeCoverageReportForAll)'=='true' and '$(CodeCoverageEnabled)'=='true' and '$(GenerateVisitedReport)' == 'true'">true</GenerateVisitedMethodsReport>
+
+    <!-- We need to filter the data to only the assembly being tested. Otherwise we will gather tons of data about other assemblies. -->
+    <AssemblyBeingTestedName>$(MSBuildProjectName.Replace('.Tests', '').Replace('.Functional', '').Replace('.Performance', '').Replace('.Unit', ''))</AssemblyBeingTestedName>
```

Comment:
You're asking why $(AssemblyBeingTestedName) is lower case in your example above? It's lower case because you're doing `Path.GetFileName("c:\some\path\dir.props")`. This function does not consult the disk so it retains the case you gave it.

I'm not sure what you're trying to achieve since it's presumably not actually to generate the string "dir.props" ?